lib/meals.js

```diff
@@ -1,5 +1,58 @@
 'use strict';
 
-module.exports = {
-  // your code goes here
+let user = {
+    name: 'Jake',
+    age: '20',
+    calorieTarget: '1800',
+    meals: [
+      {
+        title: 'Breakfeast',
+        date: '2016-06-25',
+        description: 'some good breakfeast',
+        calories: 375
+      },
+      {
+        title: 'lunch',
+        date: '2016-06-25',
+        description: 'some good lunch',
+        calories: 350
+      },
+      {
+        title: 'Dinner',
+        date: '2016-06-26',
+        description: 'some good dinner',
+        calories: 750
+      }
+    ],
+    // calculates total number of calories consumed on this date
+    caloriesEatenOn: function (myDate) {
+      let total = 0;
+      for (let i = 0; i < this.meals.length; i++){ // stepping through meals
+        if(this.meals[i].date === myDate) {        // does the date of the meal === argument passed
+          total += this.meals[i].calories;         // if so - added calories to total
+      }
+    }
+      return total;
+    },
+    // calculates the average number of calories consumed per day
+    avgDailyCalories: function (myDate) {
+      let days = 0;
+      let caloriesEats = this.caloriesEatenOn(myDate); // call caloriesEatenOn store value in caloriesEats
+      for (let i = 0; i < this.meals.length; i++) {    // stepping though meals
+          if(this.meals[i].date === myDate) {          // does date of meal === myDate (argument passed)
+          days += 1;                                   // if so - increment days by 1
+        }
+    }
```

Comment:
@apalmer0 and @nwilcox11 you might want to compare your different approaches here. Interesting to think about the differences in your thought processes while programming.